test/linalg/diagonal.jl

```diff
@@ -43,6 +43,10 @@ for relty in (Float32, Float64, BigFloat), elty in (relty, Complex{relty})
     end
 
     debug && println("Simple unary functions")
+    for op in (-,)
+        @test_approx_eq op(D) op(DM)
```

Comment:
and even then, would you be checking that you correctly negated zeros ?

I'll show myself out